ibis/expr/operations/arrays.py

```diff
@@ -103,6 +103,7 @@ class ArrayFilter(Value):
 @public
 class Unnest(Value):
     arg: Value[dt.Array]
+    offset: Optional[bool] = False
```

Comment:
```suggestion
    offset: bool
```

`Optional[T]` is to indicate that a value of type `T` can be `None` or a instance of `T`, not whether the argument itself must be be specified.

It also doesn't need to be optional in the `Unnest` operation. That optionality should happen at the user API level.